src/test/java/org/datadog/jmxfetch/TestGCMetrics.java

```diff
@@ -0,0 +1,182 @@
+            assertGCMetric(actualMetrics, "jvm.gc.minor_collection_count", "G1 Young Generation", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.minor_collection_time", "G1 Young Generation", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.major_collection_count", "G1 Old Generation", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.major_collection_time", "G1 Old Generation", "counter");
+        }
+    }
+
+    @Test
+    public void testDefaultNewGCMetricsUseZGC() throws IOException {
+        try (final MisbehavingJMXServer server = new MisbehavingJMXServer(
+                MisbehavingJMXServer.JDK_17,
+                "-XX:+UseZGC -Xmx128M -Xms128M")) {
+            final List<Map<String, Object>> actualMetrics = startAngGetMetrics(server, true);
+            assertThat(actualMetrics, hasSize(13));
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_pauses_collection_count", "ZGC Pauses", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_pauses_collection_time", "ZGC Pauses", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_cycles_collection_count", "ZGC Cycles", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_cycles_collection_time", "ZGC Cycles", "counter");
+        }
+    }
+
+    private List<Map<String, Object>> startAngGetMetrics(final MisbehavingJMXServer server, final boolean newGCMetrics) throws IOException {
+        server.start();
+        this.initApplicationWithYamlLines(
+            "init_config:",
+            "  is_jmx: true",
+            "  new_gc_metrics: " + newGCMetrics,
+            "",
+            "instances:",
+            "    -   name: jmxint_container",
+            "        host: " + server.getIp(),
+            "        collect_default_jvm_metrics: true",
+            "        max_returned_metrics: 300000",
+            "        port: " + server.getRMIPort());
+        // Run one iteration first
+        this.app.doIteration();
+        // And then pull get the metrics or else reporter does not have correct number of metrics
+        ((ConsoleReporter) appConfig.getReporter()).getMetrics();
+
+        // Actual iteration we care about
+        this.app.doIteration();
+        return ((ConsoleReporter) appConfig.getReporter()).getMetrics();
+    }
+
+    private static void assertGCMetric(final List<Map<String, Object>> actualMetrics,
+        final String expectedMetric,
+        final String gcGeneration,
+        final String metricType) {
+        MetricsAssert.assertMetric(
+            expectedMetric,
+            -1,
+            -1,
+            10.0,
+            Collections.singletonList(String.format("name:%s", gcGeneration)),
+            Arrays.asList("instance:jmxint_container", "jmx_domain:java.lang", "type:GarbageCollector"),
+            5,
+            metricType,
+            actualMetrics);
+    }
+
+    private static void assertGCMetric(final List<Map<String, Object>> actualMetrics,
```

Comment:
can this be implemented in terms of the above `assertGCMetric`? It looks like most of the tests use the above version